src/API/Site/Controllers/Assets/AssetSuggestionsController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AssetSuggestionsService;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AssetSuggestionsController
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets
+ */
+class AssetSuggestionsController extends BaseController {
+
+	/**
+	 * Service used to populate ads suggestions data.
+	 *
+	 * @var AssetSuggestionsService
+	 */
+	protected $asset_group;
+
+	/**
+	 * AssetSuggestionsController constructor.
+	 *
+	 * @param RESTServer              $server
+	 * @param AssetSuggestionsService $asset_suggestion
+	 */
+	public function __construct( RESTServer $server, AssetSuggestionsService $asset_suggestion ) {
+		parent::__construct( $server );
+		$this->asset_group = $asset_suggestion;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'assets/page/suggestions',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_pages_suggestions_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				'schema' => $this->get_api_response_schema_callback(),
+			]
+		);
+	}
+
+	/**
+	 * Get the callback function for the list of final-url suggestions request.
+	 *
+	 * @return callable
+	 */
+	protected function get_pages_suggestions_callback(): callable {
+		return function( Request $request ) {
+				return array_map(
+					function( $page ) use ( $request ) {
+						$data = $this->prepare_item_for_response( $page, $request );
+						return $this->prepare_response_for_collection( $data );
+					},
+					$this->asset_group->get_page_suggestions()
+				);
+		};
+	}
+
+
+	/**
+	 * Get the item schema for the controller.
+	 *
+	 * @return array
+	 */
+	protected function get_schema_properties(): array {
+		return [
+			'id'        => [
+				'type'        => 'number',
+				'description' => __( 'Post ID or Term ID', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+			],
+			'type'      => [
+				'type'        => 'string',
+				'description' => __( 'Post or term', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+				'enum'        => [ 'post', 'term' ],
+				'readonly'    => true,
+			],
+			'post_type' => [
+				'type'        => 'string',
+				'description' => __( 'The post type', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+				'readonly'    => true,
+			],
```

Comment:
❓ Reading a bit the project thread and explorations, as well as checking Figma file. I don't see the need of this 2 fields besides the need of future calls to other endpoints for getting the assets (I guess that would be different for post and taxonomies???). Is that correct?

Btw, maybe `type` could be computed and hence omitted from the API. For example, if `post_type` is not NULL, then is type `post`, otherwise is a `term`

Reply:
>❓ Reading a bit the project thread and explorations, as well as checking Figma file. I don't see the need of this 2 fields besides the need of future calls to other endpoints for getting the assets (I guess that would be different for post and taxonomies???). Is that correct?

Correct, to fetch the assets from a specific post or term I will need to know the type because they work differently.

I removed the post_type as it is not required anymore from the fronted perspective.  https://github.com/woocommerce/google-listings-and-ads/pull/1784/commits/f79f190ba63471b729cce93407396ea242d4b42a